jobs/build/build-sync/build.groovy

```diff
@@ -90,6 +90,49 @@ ${excludeArchesParam}
 ${dryRunParams}
 """
     }
+
+    if (params.ASSEMBLY == "stream" && !params.DOOZER_DATA_GITREF) {
+        slackChannel = slacklib.to(params.VERSION)
+        slackChannel.task("Synchronizing ART CoreOS with CI for ${params.VERSION}") {
+            buildlib.oc("${logLevel} --kubeconfig ${buildlib.ciKubeconfig} registry login")
+            def (major, minor) = commonlib.extractMajorMinorVersionNumbers(params.BUILD_VERSION)
+            // Starting with 4.12, ART is responsible for populating the CI imagestream (-n ocp is/4.12) with
+            // references to the latest machine-os-content, rhel-coreos-8, rhel-coreos-8-extensions (and
+            // potentially more with rhel9). If this is failing, it must be treated as a priority since
+            // CI will begin falling being nightly CoreOS content.
+            if ( major > 4 || minor >= 12 ) {
+                def arches = buildlib.branch_arches(groupParam)
+                // Gather a list of tags to mirror to the CI imagestream. This will include rhel-coreos-* to pick up
+                // any future RHCOS RHEL versions (e.g. rhel-coreos-9).
+                def tags = commonlib.shell(
+                    script: """
+                    oc --kubeconfig ${buildlib.ciKubeconfig} get -n ocp is/${params.BUILD_VERSION}-art-latest -o=json | jq .spec.tags[].name -r | grep -e machine-os-content -e rhel-coreos-
```

Comment:
We will want a more principled thing than `grep` here in the future - this relates to things like https://github.com/openshift/doozer/blob/3e755229ca4e00c4fd34e968a56518102ab32b6b/doozerlib/rhcos.py#L20

too.

But, seems OK to me for now.